src/main/kotlin/de/bund/digitalservice/useid/tracking/matomo/MatomoTrackingService.kt

```diff
@@ -20,24 +20,39 @@ import kotlin.text.Charsets.UTF_8
  * Documentation about matomo events
  * https://matomo.org/guide/reports/event-tracking/
  */
-@Profile("!local")
 @Service
 class MatomoTrackingService(trackingProperties: TrackingProperties, private val webRequests: WebRequests) {
 
     private val log = KotlinLogging.logger {}
     private val siteId = trackingProperties.matomo.siteId
     private val domain = trackingProperties.matomo.domain
+    private val dimensionIdTenant = trackingProperties.matomo.dimensionIdTenant
 
     fun constructEventURL(e: MatomoEvent): String {
         val session = e.sessionId?.let { "&uid=$it" } ?: ""
-        val userAgent = e.userAgent?.let { "&ua=${UriUtils.encode(e.userAgent, UTF_8)}" } ?: ""
+        val userAgent = e.userAgent?.let { "&ua=${encode(e.userAgent)}" } ?: ""
         // custom dimension for tenantId -> https://matomo.org/faq/reporting-tools/create-track-and-manage-custom-dimensions/
-        val tenantId = e.tenantId?.let { "&dimension3=${UriUtils.encode(e.tenantId, UTF_8)}" } ?: ""
-        val url = "$domain?idsite=$siteId&rec=1&ca=1&e_c=${e.category}&e_a=${e.action}&e_n=${e.name}$session$userAgent$tenantId"
-        log.debug { url }
-        return url
+        val tenantId = e.tenantId?.let { "&dimension$dimensionIdTenant=${encode(e.tenantId)}" } ?: ""
+
+        return UriComponentsBuilder
+            .newInstance()
+            .scheme("https")
+            .host(domain)
+            .path("?idsite=$siteId")
```

Comment:
Das praktische am `UriComponentsBuilder` ist gerade, dass wir uns nicht selbst Gedanken darum machen müssen, ob `?` oder `&`, dafür gibt es `.queryParam()`. Und für optionale sollte `.queryParamIfPresent()` auch helfen, sodass wir das gar nicht mehr selbst machen müssen oben.

Reply:
UriComponentBuilder encoded query parameter nicht. Es ist also für unsere Zwecke nicht richtig einsetzbar. 
See for example [here](https://stackoverflow.com/questions/18138011/url-encoding-using-the-new-spring-uricomponentsbuilder)
Ich versuche trotzdem was gutes zu bauen